pex/resolve/lock_resolver.py

```diff
@@ -0,0 +1,289 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import
+
+import functools
+import hashlib
+from collections import OrderedDict
+from multiprocessing.pool import ThreadPool
+
+from pex.common import pluralize
+from pex.compatibility import cpu_count
+from pex.fetcher import URLFetcher
+from pex.network_configuration import NetworkConfiguration
+from pex.orderedset import OrderedSet
+from pex.pip import PackageIndexConfiguration
+from pex.resolve import lockfile
+from pex.resolve.locked_resolve import Artifact, DownloadableArtifact, Resolved
+from pex.resolve.lockfile import parse_lockable_requirements
+from pex.resolve.lockfile.download_manager import DownloadedArtifact, DownloadManager
+from pex.resolve.requirement_configuration import RequirementConfiguration
+from pex.resolve.resolver_configuration import ResolverVersion
+from pex.resolve.resolvers import Installed
+from pex.resolver import BuildAndInstallRequest, BuildRequest, InstallRequest
+from pex.result import Error, ResultError, catch, try_
+from pex.targets import Target, Targets
+from pex.tracer import TRACER
+from pex.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Dict, Iterable, Optional, Sequence, Union
+
+
+class URLFetcherDownloadManager(DownloadManager):
+    _BUFFER_SIZE = 65536
```

Comment:
I hadn't seen this number used before but knew you chose it for a reason. Fine if you don't want to burn a tree, but otherwise this would help newbs like me

```suggestion
    _BUFFER_SIZE = 65536  # i.e. max number of 16 bits.
```

Reply:
Yeah - not the reason. I can circle back, but, as always, buffer size selection is all about perf and nothing more. The number of bits may tangentially relate to the size of underlying os buffers, but its not the point, the point, hopefully, is you measured on some reasonably diverse set of inputs and found a best-fit. That's what this was.